src/core/hooks/InvitationsGraphQL/useUpdateInvitation.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback } from 'react';
+
+import { IInvitationDataProps, ResponseStatus } from 'context';
+import { allInvitationsQuery } from 'core/operations';
+import { useUpdateInvitationMutation } from 'core/graphql';
+
+export interface IUpdateInvitation {
+  invitations: IInvitationDataProps[],
+  companyId: string,
+  status: ResponseStatus
+}
+
+export const useUpdateInvitation = (): [(props: IUpdateInvitation) => void] => {
+  const [ updateInvitation ] = useUpdateInvitationMutation({
+    refetchQueries: [{
+      query: allInvitationsQuery
+    }]
+  });
+
+  const onChangeInvitationStatusClick = useCallback(async (props: IUpdateInvitation) => {
+    const { invitations, companyId, status } = props;
+
+    const findInvitation = await invitations.find(({company}) => company.id === companyId);
+
+    await updateInvitation({
```

Comment:
It is better to not use async/await, because you should have at least `loading` when using `useMutation`: [link](https://www.apollographql.com/docs/react/data/mutations/#executing-a-mutation)